crates/swc_css_parser/src/parser/at_rule.rs

```diff
@@ -1353,105 +1391,161 @@ where
+        };
+
+        Ok(PageSelectorList {
+            span: Span::new(start_pos, last_pos, Default::default()),
+            selectors,
+        })
+    }
+}
+
 impl<I> Parse<PageSelector> for Parser<I>
 where
     I: ParserInput,
 {
     fn parse(&mut self) -> PResult<PageSelector> {
         self.input.skip_ws()?;
 
-        let start = self.input.cur_span()?.lo;
+        let span = self.input.cur_span()?;
 
-        let ident = if is!(self, Ident) {
+        let page_type = if is!(self, Ident) {
             Some(self.parse()?)
         } else {
             None
         };
 
-        let pseudo = if eat!(self, ":") {
-            Some(self.parse()?)
+        let pseudos = if is!(self, ":") {
+            let mut pseudos = vec![];
+
+            loop {
+                if !is!(self, ":") {
+                    break;
+                }
+
+                let pseudo = self.parse()?;
+
+                pseudos.push(pseudo);
+            }
+
+            Some(pseudos)
         } else {
             None
         };
 
         Ok(PageSelector {
-            span: span!(self, start),
-            ident,
-            pseudo,
+            span: span!(self, span.lo),
+            page_type,
+            pseudos,
         })
     }
 }
 
-impl<I> Parse<PageRuleBlock> for Parser<I>
+impl<I> Parse<PageSelectorType> for Parser<I>
 where
     I: ParserInput,
 {
-    fn parse(&mut self) -> PResult<PageRuleBlock> {
+    fn parse(&mut self) -> PResult<PageSelectorType> {
         let span = self.input.cur_span()?;
-        expect!(self, "{");
-        self.input.skip_ws()?;
-        let mut items = vec![];
-
-        if !is!(self, "}") {
-            loop {
-                self.input.skip_ws()?;
-
-                let q = self.parse()?;
-                items.push(q);
-
-                self.input.skip_ws()?;
-
-                if is_one_of!(self, EOF, "}") {
-                    break;
-                }
-            }
-        }
-
-        expect!(self, "}");
+        let value = self.parse()?;
 
-        Ok(PageRuleBlock {
+        Ok(PageSelectorType {
             span: span!(self, span.lo),
-            items,
+            value,
         })
     }
 }
 
-impl<I> Parse<PageRuleBlockItem> for Parser<I>
+impl<I> Parse<PageSelectorPseudo> for Parser<I>
 where
     I: ParserInput,
 {
-    fn parse(&mut self) -> PResult<PageRuleBlockItem> {
-        match cur!(self) {
-            Token::AtKeyword { .. } => Ok(PageRuleBlockItem::Nested(self.parse()?)),
-            _ => {
-                let p = self
-                    .parse()
-                    .map(Box::new)
-                    .map(PageRuleBlockItem::Declaration)?;
-                eat!(self, ";");
+    fn parse(&mut self) -> PResult<PageSelectorPseudo> {
+        let span = self.input.cur_span()?;
+
+        expect!(self, ":");
 
-                Ok(p)
+        let value = match cur!(self) {
+            Token::Ident { value, .. }
+                if &*value.to_ascii_lowercase() == "left"
```

Comment:
This seems ineffcient.
We can do `matches!(&*value.to_ascii_lowercase(), "left" | "right" | "first" | "blank")` instead.

Reply:
Yep, I will do it firstly when rebase this PR we already need this in many places